ports/esp32/machine_pwm.c

```diff
@@ -241,10 +241,13 @@ static void set_freq(machine_pwm_obj_t *self, unsigned int freq, ledc_timer_conf
         // Configure the new resolution and frequency
         timer->duty_resolution = res;
         timer->freq_hz = freq;
-        #if SOC_LEDC_SUPPORT_XTAL_CLOCK
-        timer->clk_cfg = LEDC_USE_XTAL_CLK;
-        #else
+
+        #if SOC_LEDC_SUPPORT_PLL_DIV_CLOCK
+        timer->clk_cfg = LEDC_USE_PLL_DIV_CLK;
+        #elif SOC_LEDC_SUPPORT_APB_CLOCK
         timer->clk_cfg = LEDC_USE_APB_CLK;
+        #elif SOC_LEDC_SUPPORT_XTAL_CLOCK
+        timer->clk_cfg = LEDC_USE_XTAL_CLK;
```

Comment:
This changes the behaviour for all chips except for ESP32, I think? but probably in a good way... as it breaks down as:

```
❯ rg -g soc_caps.h "SOC_LEDC_SUPPORT_.+_CLOCK" components/soc
components/soc/esp32h2/include/soc/soc_caps.h
267:#define SOC_LEDC_SUPPORT_PLL_DIV_CLOCK      (1)
268:#define SOC_LEDC_SUPPORT_XTAL_CLOCK         (1)

components/soc/esp32c2/include/soc/soc_caps.h
166:#define SOC_LEDC_SUPPORT_PLL_DIV_CLOCK  (1)
167:#define SOC_LEDC_SUPPORT_XTAL_CLOCK     (1)

components/soc/esp32/include/soc/soc_caps.h
231:#define SOC_LEDC_SUPPORT_APB_CLOCK       (1)

components/soc/esp32s2/include/soc/soc_caps.h
223:#define SOC_LEDC_SUPPORT_APB_CLOCK       (1)
225:#define SOC_LEDC_SUPPORT_XTAL_CLOCK      (1)

components/soc/esp32c3/include/soc/soc_caps.h
219:#define SOC_LEDC_SUPPORT_APB_CLOCK       (1)
220:#define SOC_LEDC_SUPPORT_XTAL_CLOCK      (1)

components/soc/esp32s3/include/soc/soc_caps.h
227:#define SOC_LEDC_SUPPORT_APB_CLOCK       (1)
228:#define SOC_LEDC_SUPPORT_XTAL_CLOCK      (1)

components/soc/esp32c6/include/soc/soc_caps.h
267:#define SOC_LEDC_SUPPORT_PLL_DIV_CLOCK      (1)
268:#define SOC_LEDC_SUPPORT_XTAL_CLOCK         (1)

components/soc/esp32p4/include/soc/soc_caps.h
274:#define SOC_LEDC_SUPPORT_PLL_DIV_CLOCK      (1)
275:#define SOC_LEDC_SUPPORT_XTAL_CLOCK         (1)
```

Which I read as:

* ESP32 only supports APB clock source.
* All chips apart from ESP32 support both XTAL clock and one of APB or PLL_DIV.

And:

* In current master all chips apart from ESP32 get the slower XTAL clock source (40MHz).
* With this PR they now get either APB or PLL_DIV, whichever is supported. (I believe 80MHz or 96MHz, respectively.)

However there is a bunch of (uncommented) code earlier in this function which looks wrong - i.e. it calculates the duty resolution, but it calculates it from the assumption the source clock is 80MHz. This is only true on ESP32 on current master, plus now true for some other chips in this PR.

ESP-IDF driver provides a function `ledc_find_suitable_duty_resolution`, and the function doc says to find the source clock frequency via `esp_clk_tree_src_get_freq_hz` - so probably all that earlier code can be replaced with calls to these two functions?

We can potentially leave this for a future PR, though. As I think this PR already makes the behaviour more  correct, not less correct...